File: app/src/main/java/com/designpatterns/demo/behavioral/template/java/CDAccount.java
package com.designpatterns.demo.behavioral.template.java;

/**
 * 具体模板角色类
 *
 * @author zhangzhiguo
 * @version 1.0.0
 * @email: 810903987@qq.com
 * @date 2018/6/7 上午11:42
 * @project_name DesignPatternsDemo
 */
public class CDAccount extends Account {
    @Override
    protected String doCalculateAccountType() {
        return "Certificate of Deposite";
    }

    @Override
    protected double doCalculateInterestRate() {
        return 0.06;
    }
}

File: app/src/main/java/com/designpatterns/demo/behavioral/template/kotlin/CDAccount.kt
package com.designpatterns.demo.behavioral.template.kotlin

/**
 * 具体模板角色类
 *
 * @author  zhangzhiguo
 * @email: 810903987@qq.com
 * @date 2018/6/7 下午1:43
 * @project_name DesignPatternsDemo
 * @version 1.0.0
 *
 */
class CDAccount : Account() {

    override fun doCalculateAccountType(): String = "Certificate of Deposite"

    override fun doCalculateInterestRate(): Double = 0.06

}
